test/unit/collector/serverless.test.js

```diff
@@ -175,11 +179,112 @@ describe('ServerlessCollector API', () => {
     it('compresses full payload and writes formatted to stdout', (done) => {
       api.payload = {type: 'test payload'}
       api.flushPayload(() => {
-        const logPayload = JSON.parse(logStub.args[0][0])
+        let logPayload = null
+
+        logPayload = JSON.parse(logStub.args[0][0])
+
+        expect(logPayload).to.be.an('array')
+        expect(logPayload[0]).to.be.a('number')
+
+        expect(logPayload[1]).to.equal('NR_LAMBDA_MONITORING')
+        expect(logPayload[2]).to.be.a('string')
+
+        done()
+      })
+    })
+    it('handles very large payload and writes formatted to stdout', done => {
+      api.payload = {type: 'test payload'}
+      for (let i = 0; i < 4096; i++) {
+        api.payload[`customMetric${i}`] = Math.floor(Math.random() * 100000)
+      }
+
+      api.flushPayload(() => {
+        let logPayload = null
+
+        if (shouldUsePipe) {
+          logPayload = JSON.parse(logStub.getCall(0).args[1])
```

Comment:
Seems like this path might not work but tis related to the forking which I'm hoping we can get rid off.

Reply:
I've refactored that out.